### PR TITLE
Create decorator classes and implement composition

### PR DESCRIPTION
**Pull request by lenkon**
#### The following are implemented:
- Interface Nameable class created.
- Made the Person class inherit the Nameable class.
- Created BaseDecorator class that inherits the Nameable class.
- Created CapitalizeDecorator class that inherits BaseDecorator class and which capitalizes a string.
- Created a TrimmerDecorator class that inherits BaseDecorator and which trims a string.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0...10]
```
Comment (VICTOR-CHUKWU) [inline trimmer_decorator.rb]: - [x] According to the project requiremnet you should trim to a maximum of 10, currently, you are timing to a maximum of 11 characters, so kindly change this to  ` @nameable.correct_name[0...9]`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @lenkon  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (lenkon) [reply trimmer_decorator.rb]: Hi @VICTOR-CHUKWU,

Thanks for the feedback. I have updated the code accordingly.

Best Regards,
K. Inoulen Shanongba

**Review by DammyShittu — APPROVED** [review]
Hi @lenkon,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**